### PR TITLE
Add monitor dialogue for Spark on ADL

### PR DESCRIPTION
**Pull request by t-rufang**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/serverexplore/ui/SparkServerlessClusterMonitorDialog.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    private static final int REFRESH_INTERVAL = 1;
+
+    public SparkServerlessClusterMonitorDialog(@NotNull SparkServerlessClusterNode clusterNode,
+                                               @NotNull AzureSparkServerlessCluster cluster) {
+        super((Project)clusterNode.getProject(), true);
+        this.ctrlProvider = new SparkServerlessClusterStatesCtrlProvider(
+                this, new IdeaSchedulers((Project)clusterNode.getProject()), cluster);
+        this.cluster = cluster;
+
+        livyEndpointHyperLink.addActionListener(event -> BrowserUtil.browse(livyEndpointHyperLink.getURI()));
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/serverexplore/ui/SparkServerlessClusterMonitorDialog.java]: This action listener can be JXHyperLinkWithUri's default action

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/serverexplore/ui/SparkServerlessClusterMonitorDialog.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+        init();
+        this.setTitle("Cluster Status");
+        this.setModal(true);
+    }
+
+    @Override
+    public void show() {
+        refreshSub = ctrlProvider.updateAll()
+                .subscribeOn(Schedulers.io())
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/serverexplore/ui/SparkServerlessClusterMonitorDialog.java]: Could you try to remove this line?

File: Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/SparkServerlessClusterStatesModel.java [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+import java.net.URI;
+
+public class SparkServerlessClusterStatesModel implements Cloneable{
+    @NotNull
+    private String masterState;
+    @NotNull
+    private String workerState;
+
+    @NotNull
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/SparkServerlessClusterStatesModel.java]: No need for `int` value

File: Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/SparkServerlessClusterStatesModel.java [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    private int masterFailed;
+    @NotNull
+    private int workerFailed;
+
+    @NotNull
+    private int masterOutstanding;
+    @NotNull
+    private int workerOutstanding;
+
+    @NotNull
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/SparkServerlessClusterStatesModel.java]: `@Nullable`? Or you need to new URI() here.

File: Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/SparkServerlessClusterStatesModel.java [hunk 5]
```diff
@@ -0,0 +1,201 @@
+package com.microsoft.azure.sparkserverless.serverexplore;
+
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+
+import java.net.URI;
+
+public class SparkServerlessClusterStatesModel implements Cloneable{
+    @NotNull
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/SparkServerlessClusterStatesModel.java]: `@Nullable`? Or you need to assign an empty string here.

File: Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java [hunk 6]
```diff
@@ -79,6 +79,8 @@ protected void loadActions() {
         String suffix = "/?adlaAccountName=" + adlAccount.getName();
         addAction("Delete", new SparkServerlessDestroyAction(
                 this, cluster, adlAccount, SparkServerlessClusterOps.getInstance().getDestroyAction()));
+        addAction("View Cluster Status", new SparkServerlessMonitorAction(
+                this, cluster, SparkServerlessClusterOps.getInstance().getMonitorAction()));
         addAction("Open Livy UI", new NodeActionListener() {
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java]: Need to remove this one.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/common/JXHyperLinkWithUri.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
+package com.microsoft.azure.sparkserverless.common;
+
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import org.jdesktop.swingx.JXHyperlink;
+import org.jdesktop.swingx.hyperlink.HyperlinkAction;
+
+import java.net.URI;
+
+public class JXHyperLinkWithUri extends JXHyperlink {
+    private URI uri;
```
Comment (t-rufang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/common/JXHyperLinkWithUri.java]: `@Nullable`

**Pull request merged**